### ISSUE TITLE
log: wrong identity in syslog on 2.1+

### ISSUE THREAD

**Issue by Totktonada** [tarantool/tarantool#4785]
Tarantool version: 2.0.4-837-g5db765a76.
OS version: Linux.

After update from 1.10 to 2.3.1.28 a customer found that indentity column in syslog (/var/log/messages) becomes a date:

```

2020-02-17T19:50:01.242 o-hb-shinobi1   [systemd]       <info>  {daemon}        Starting Session 119151 of user root.
--
2020-02-17T19:50:01.248 o-hb-shinobi1   [systemd]       <info>  {daemon}        Started Session 119156 of user root.
2020-02-17T19:50:01.252 o-hb-shinobi1   [systemd]       <info>  {daemon}        Starting Session 119156 of user root.
2020-02-17T19:50:03.916 o-hb-shinobi1   [systemd]       <info>  {daemon}        Removed slice User Slice of nobody.
2020-02-17T19:50:03.917 o-hb-shinobi1   [systemd]       <info>  {daemon}        Stopping User Slice of nobody.
2020-02-17T19:50:06.892 o-hb-shinobi1   [2020-02-17]    <notice>        {user}  19:50:06.892 [28634] main/136/console/unix/: I> queue: disconnected unix/:(socket), sid=71, fid=136
2020-02-17T19:50:06.892 o-hb-shinobi1   [2020-02-17]    <notice>        {user}  19:50:06.892 [28634] main/136/console/unix/: I> disconnected unix/:(socket), sid=71, fid=136
2020-02-17T19:50:09.236 o-hb-shinobi1   [2020-02-17]    <notice>        {user}  19:50:09.236 [28634] main/137/console/unix/: I> connected unix/:(socket), sid=73, fid=137
2020-02-17T19:50:09.243 o-hb-shinobi1   [2020-02-17]    <notice>        {user}  19:50:09.243 [28634] main/137/console/unix/: I> queue: disconnected unix/:(socket), sid=73, fid=137
2020-02-17T19:50:09.243 o-hb-shinobi1   [2020-02-17]    <notice>        {user}  19:50:09.243 [28634] main/137/console/unix/: I> disconnected unix/:(socket), sid=73, fid=137
2020-02-17T19:50:29.722 o-hb-shinobi1   [2020-02-17]    <notice>        {user}  19:50:29.722 [28634] main/110/main I> connected 10.161.133.2:15594, sid=74, fid=110
2020-02-17T19:50:29.743 o-hb-shinobi1   [2020-02-17]    <notice>        {user}  19:50:29.743 [28634] main/110/main I> queue: disconnected 10.161.133.2:15594, sid=74, fid=110
2020-02-17T19:50:29.743 o-hb-shinobi1   [2020-02-17]    <notice>        {user}  19:50:29.744 [28634] main/110/main I> disconnected 10.161.133.2:15594, sid=74, fid=110
2020-02-17T19:50:59.819 o-hb-shinobi1   [2020-02-17]    <notice>        {user}  19:50:59.819 [28634] main/110/main I> connected 10.161.133.2:15596, sid=75, fid=110
```

box configuration is the following:

```
log_level: 5
log_nonblock: false
log: syslog:identity=tarantool,server=127.0.0.1:514
log_format: plain
```

How to reproduce:

```
$ ./src/tarantool
tarantool> box.cfg{log = 'syslog:identity=tarantool'}
tarantool> require('log').info('aaa')
tarantool> Ctrl+D
$ tail /var/log/messages
```

Caused by [2.0.4-837-g5db765a76](https://github.com/tarantool/tarantool/commit/5db765a76ab7270b350960a2dd6d89dcd3bdc032). The following code was removed from `say_set_log_format()` and it seems this is the reason of the problem:

```c
        /*   
         * For syslog, default or boot log type the log format can
         * not be changed.
         */
        bool allowed_to_change = log_default->type == SAY_LOGGER_STDERR ||
                                 log_default->type == SAY_LOGGER_PIPE ||
                                 log_default->type == SAY_LOGGER_FILE;
        <...>
        case SF_PLAIN:
                if (!allowed_to_change) {
                        return;
                }    
                log_set_format(log_default, say_format_plain);
                break;
```

Jira: TNT-2.

BTW, can we, say, add an errinj to write into a custom location for syslog in order to test it?

**Comment by OKriw** [comment]
I was too brave and blind - I have overlooked setting format later in say_logger_init  after log_syslog_init.

**Comment by Totktonada** [comment]
Fixed in [2.5.0-2-g098324556](https://github.com/tarantool/tarantool/commit/098324556aadf9f7e6b41758fbb62cc91f32d284), [2.4.1-2-g018cf41bb](https://github.com/tarantool/tarantool/commit/018cf41bb75951b0cfafe80ba909593bf35b79c4), [2.3.2-2-g285a25b8d](https://github.com/tarantool/tarantool/commit/285a25b8de9fdbcc13fbd455d43f9af148dcf377).

2.1 and 2.2 are EOL. This means that there will no new release for those branches. This change is not pushed to them.

**Comment by Totktonada** [comment]
(GitHub [experiences](https://www.githubstatus.com/incidents/dsf2qtzh4jpz) a problem, so the issue is not closed from the pushed commit. I'll close it manually.)
